ckanext/canada/tables/contracts.yaml

```diff
@@ -1786,6 +1781,9 @@ iii. Quand un contrat porte sur plus d’un article économique, il est recomman
           END IF;
         END IF;
 
+        IF length(trim(both E'\t\n\x0b\x0c\r ' from NEW.economic_object_code)) > 4 THEN
+          errors := errors || ARRAY[['economic_object_code', 'This field is limited to only 3 or 4 digits.']];
+        END IF;
```

Comment:
trim the value in NEW.economic_object_code so that we don't store any white-space in the db either